src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,169 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MagicWandService {
+  /**4-Way floodfill (left, right, up, down):
+   * Return set of coordinates(formatted as a 1-D array).
+   * Coordinates correspond to pixels considered part of the mask.
+   */
+  /* tslint:disable */
+  floodfill(imgData: ImageData, xCoord: number, yCoord: number,
+            tolerance: number): Set<number> {
+    // Use basic queue to immitate recursive approach of the stack
+    // possible TODO: set array size after mvp to optimize heap memory access
+    const visit: Array<Array<number>> = new Array();
+    const visited: Set<number> = new Set();
+    // Use a set for mask; mainly do iter and set operations on masks
+    const mask: Set<number> = new Set();
+    // Represent [R,G,B,A] attributes of initial pixel
+    const originalPixel: Array<number> = this.dataArrayToRGBA(imgData, xCoord, yCoord);
+
+    visit.push([xCoord, yCoord]);
+    // Convert [x,y] format coord to 1-D equivalent of imgData.data (DataArray)
+    let indexAsDataArray: number = this.coordToDataArrayIndices(xCoord, yCoord, imgData.width)[0];
+    visited.add(indexAsDataArray);
+
+    // Loop until no more adjacent pixel's within tolerance level
+    while(visit.length != 0) {
+      let coord: Array<number> = visit.pop();
+      // Unpack coord
+      let x: number = coord[0];
+      let y: number = coord[1];
+
+      // Operational part of while-loop
+      mask.add(this.coordToDataArrayIndices(x, y, imgData.width)[0]);
+
+      // Loop part of while-loop
+
+      // Get coords of adjacent pixels
+      let leftPixelCoord: Array<number> = [x - 1, y];
+      let rightPixelCoord: Array<number> = [x + 1, y];
+      let upPixelCoord: Array<number> = [x, y - 1];
+      let downPixelCoord: Array<number> = [x, y + 1];
+      // Add coords of adjacent pixels to the heap
+      let isStillMask: boolean = this.getIsMask(originalPixel, imgData, leftPixelCoord, tolerance, visited);
+      // Check left pixel
+      if (isStillMask) {
+        visit.push(leftPixelCoord);
+      }
```

Comment:
You want to avoid cut-and-pasting code as much as possible. For example, if you ever want to change one of the four copies of this block, you'll have to change all of them. I would suggest:

let neighbors = [[x-1, y], [x+1, y], [x, y-1], [x, y+1]];
for (let neighbor of neighbors) {
 ...
}